### DISCUSSION TITLE
Adding support for VertexAI

### DISCUSSION THREAD

**Discussion by heiko-hotz** [stanfordnlp/dspy#640] [Ideas]
Google offers its Gemini models in two ways: the Gemini API and the VertexAI API. Currently only the Gemini API is supported in DSPy. The VertexAI API is the more "serious" API - it's geared towards developers who want to build production-ready systems. See also https://stackoverflow.com/questions/78007243/utilizing-gemini-through-vertex-ai-or-through-google-generative-ai

It would be great if DSPy could add support for the VertexAI API.

**Comment by pranavchaniyara** [comment]
@marshmellow77 Can you give me some info on how to use Gemini API in DSPy?

**Comment by heiko-hotz** [comment]
Hi @pranavchaniyara , you can simply use `gemini = dspy.GoogleVertexAI(model="gemini-1.5-flash")`

Here is a complete program to illustrate how to use Gemini in DSPY: https://github.com/stanfordnlp/dspy/issues/1187
